wildfly-mcp/subsystem/src/main/java/org/wildfly/extension/mcp/deployment/McpSseHandlerServiceInstaller.java

```diff
@@ -62,10 +106,72 @@ public void run() {
                 ROOT_LOGGER.endpointUnregistered(ssePath, host.get().getName());
             }
         };
-        ServiceInstaller.builder(start, stop)
-                .requires(List.of(host, ServiceDependency.on(Capabilities.MCP_SERVER_PROVIDER_DESCRIPTOR)))
-                .asActive()
-                .build()
-                .install(context);
+        if (oidcSecured) {
+            ServiceInstaller.builder(start, stop)
+                    .requires(List.of(host, ServiceDependency.on(Capabilities.MCP_SERVER_PROVIDER_DESCRIPTOR), securityDomain))
+                    .asActive()
+                    .build()
+                    .install(context);
+        } else {
+            ServiceInstaller.builder(start, stop)
+                    .requires(List.of(host, ServiceDependency.on(Capabilities.MCP_SERVER_PROVIDER_DESCRIPTOR)))
+                    .asActive()
+                    .build()
+                    .install(context);
+        }
+    }
+
+    private HttpServerAuthenticationMechanismFactory getHttpServerAuthenticationMechanismFactory(DeploymentUnit deploymentUnit) {
+        WarMetaData warMetaData = deploymentUnit.getAttachment(WarMetaData.ATTACHMENT_KEY);
+        if (warMetaData == null) {
+            return null;
+        }
+        String json = getJsonConfiguration(warMetaData);
+        if (json != null) {
+            OidcClientConfiguration oidcClientConfiguration = OidcClientConfigurationBuilder.build(new ByteArrayInputStream(json.getBytes(StandardCharsets.UTF_8)));
+            oidcClientConfiguration.setBearerOnly(true);
```

Comment:
Couldn't this be specified in the json config?

Reply:
See answer above: I'm not using the json file but creating it from the subsystem configuraiton. I wanted to make sure I was in bearer mode

Reply:
Can this be part of the subsystem configuration then? Was mostly just curious why the config is being updated separately to specify this.